### ISSUE TITLE
Caffe converter silently fails on (some) non-supported layer types

### ISSUE THREAD

**Issue by festlv** [tiny-dnn/tiny-dnn#586]
I have been trying out caffe converter example on a custom caffe model.

The prediction vector of the imported model was all zeroes. During debugging of this issue I found out that tiny-dnn network was correctly constructed, but loading weights had silently stopped after encountering BatchNorm layer- it can't be mapped to tiny-dnn layer type for weight loading.

I'd suggest at least changing this line to throw an error in this case, so that unsupported layer types trigger obvious error:
https://github.com/tiny-dnn/tiny-dnn/blob/master/tiny_dnn/io/caffe/layer_factory.h#L146

**Comment by edgarriba** [comment]
@festlv Thanks for reporting this bug! please submit a PR with the pach

**Comment by bhack** [comment]
And also covering missing layer in tiny if you need :)

**Comment by festlv** [comment]
Submitted a PR.

I do need this missing layer, but I don't know enough about internals of tiny-dnn, caffe and even how exactly batchnorm works, so I won't be able to add it.

**Comment by festlv** [comment]
The core of this issue has been fixed in #587.
The missing BatchNorm support is in #595.

I have the following suggestions regarding caffe converter:

1) List supported layer types in documentation (https://github.com/tiny-dnn/tiny-dnn/blob/master/tiny_dnn/io/caffe/layer_factory_impl.h#L857).
2) Make the inference in general more robust (maybe a debug mode with assertions). I have been able to proceed through inference although the weights were not loaded for any layers.
